mapper/thin_edge_json/src/json.rs

```diff
@@ -289,6 +298,101 @@ mod tests {
             .and_hms(0, 0, 0)
     }
 
+    #[test]
+    fn valid_short_time_stamp() {
+        let input = r#"{
+            "time" : "2021-04-30T17:03:14.123+02:00"
+        }"#;
+        let json = json::parse(&input).unwrap();
+
+        let expected_output = FixedOffset::east(2 * 3600)
+            .ymd(2021, 04, 30)
+            .and_hms_milli(17, 03, 14, 123);
+
+        let time = json.entries().nth(0).unwrap();
+        let output = ThinEdgeJson::check_timestamp_for_iso8601_complaint(time.1).unwrap();
+        assert_eq!(output, expected_output);
+    }
+
+    #[test]
+    fn valid_string_time_stamp() {
+        let input = r#"{
+            "time" : "2021-04-30T17:03:14.123456789+02:00"
+        }"#;
+
+        let json = json::parse(&input).unwrap();
+        let expected_output = FixedOffset::east(2 * 3600)
+            .ymd(2021, 04, 30)
+            .and_hms_nano(17, 03, 14, 123456789);
+
+        let time = json.entries().nth(0).unwrap();
+        let output = ThinEdgeJson::check_timestamp_for_iso8601_complaint(time.1).unwrap();
+        assert_eq!(output, expected_output);
+    }
+
+    #[test]
+    fn invalid_time_stamp() {
+        let input = r#"{
+            "time" : "2013-06-2217:03:14.000658767+02:00"
+        }"#;
+        let expected_error = r#"Invalid ISO8601 timestamp (expected YYYY-MM-DDThh:mm:ss.sss.±hh:mm): "2013-06-2217:03:14.000658767+02:00": input contains invalid characters"#;
+        let json = json::parse(&input).unwrap();
+        let time = json.entries().nth(0).unwrap();
+        let output = ThinEdgeJson::check_timestamp_for_iso8601_complaint(time.1).unwrap_err();
+        assert_eq!(output.to_string(), expected_error);
+    }
+
+    #[test]
+    fn check_from_json_str_short_timestamp() {

```

Comment:
Isn't the name a bit misleading? Isn't this test checking a payload without a timestamp?

Reply:
The timestamp will be passed as part of this API "ThinEdgeJson::from_str(input, timestamp).unwrap();"

Reply:
Agree with Albin, this test name give absolutely no hint on what is tested. 

* Looking closer, this test is *unrelated* to this issue. It simply tests that the timestamp provided to the `from_str()` method is used when no timestamp is provided in the text input.

Reply:
Looking even closer, this test is derived from the test example I provided. However, simply by moving the timestamp *out* of the input, you simply destroy the point of the test.

Reply:
Yes, I moved because we were also passing timestamp as part of the API call, felt its redundant. Now I put back the timestamp, may be our API design is not clearer.

Reply:
> Agree with Albin, this test name give absolutely no hint on what is tested.
> 
> * Looking closer, this test is _unrelated_ to this issue. It simply tests that the timestamp provided to the `from_str()` method is used when no timestamp is provided in the text input.

Now I have renamed both the test cases. One with check_json_str_with_millisecond_timestamp and another is check_json_str_with_nanosecond_timestamp. Hope now its more clearer